google/cloud/aiplatform/model_evaluation/model_evaluation_job.py

```diff
@@ -0,0 +1,309 @@
+    # "unstructured_without_feature_attribution": "TODO",
+    # "unstructured_with_feature_attribution": "TODO",
+}
+
+
+class ModelEvaluationJob(pipeline_based_service._VertexAiPipelineBasedService):
+
+    _template_ref = _MODEL_EVAL_PIPELINE_TEMPLATES
+
+    @property
+    def _metadata_output_artifact(self) -> Optional[str]:
+        """The resource uri for the ML Metadata output artifact from the evaluation component of the Model Evaluation pipeline"""
+        if self.state == gca_pipeline_state_v1.PipelineState.PIPELINE_STATE_SUCCEEDED:
+            for task in self.backing_pipeline_job._gca_resource.job_detail.task_details:
+                if (
+                    task.task_name.startswith("model-evaluation")
+                    and "evaluation_metrics" in task.outputs
+                ):
+                    return task.outputs["evaluation_metrics"].artifacts[0].name
+
+    def __init__(
+        self,
+        evaluation_pipeline_run: str,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Retrieves a ModelEvaluationJob and instantiates its representation.
+        Example Usage:
+            my_evaluation = aiplatform.ModelEvaluationJob(
+                pipeline_job_id = "projects/123/locations/us-central1/pipelineJobs/456"
+            )
+            my_evaluation = aiplatform.ModelEvaluationJob(
+                pipeline_job_id = "456"
+            )
+        Args:
+            evaluation_pipeline_run (str):
+                Required. A fully-qualified pipeline job run ID.
+                Example: "projects/123/locations/us-central1/pipelineJobs/456" or
+                "456" when project and location are initialized or passed.
+            project (str):
+                Optional. Project to retrieve pipeline job from. If not set, project
+                set in aiplatform.init will be used.
+            location (str):
+                Optional. Location to retrieve pipeline job from. If not set, location
+                set in aiplatform.init will be used.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials to use to retrieve this pipeline job. Overrides
+                credentials set in aiplatform.init.
+        """
+        super().__init__(
+            pipeline_job_id=evaluation_pipeline_run,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+    @staticmethod
+    def _get_template_url(data_type, feature_attributions) -> str:
+
+        template_type = data_type
+
+        if feature_attributions:
+            template_type += "_with_feature_attribution"
+        else:
+            template_type += "_without_feature_attribution"
+
+        return ModelEvaluationJob._template_ref[template_type]
+
+    @classmethod
+    def submit(
```

Comment:
(nit) can this be private method, since i believe `model.evaluate()` is the intended public API for eval?

Reply:
Yes good point, I'll update this to be a private method.

Reply:
Update: I chatted with Sasha and we decided the `ModelEvaluationJob` class should be private. Just updated this.